spinalcordtoolbox/scripts/sct_apply_transfo.py

```diff
@@ -314,17 +324,16 @@ def apply(self):
         warping_field = fname_warp_list_invert[-1]
         # If the last transformation is not an affine transfo, we need to compute the matrix space of the concatenated
         # warping field
-        if not isLastAffine and crop_reference in [1, 2]:
+        if not isLastAffine and crop_reference in [1, 2, 3]:
             printv('Last transformation is not affine.')
+            img_out = Image(fname_out)
+            # Extract only the first n dims of the warping field by creating a dummy image with the correct shape
+            img_warp = Image(warping_field)
+            warp_shape = img_warp.data.shape[:int(dim)]  # dim = {'2', '3', '4'}
+            img_warp_ndim = Image(np.ones(warp_shape), hdr=img_warp.hdr)
             if crop_reference in [1, 2]:
-                # Extract only the first ndim of the warping field
-                img_warp = Image(warping_field)
-                if dim == '2':
-                    img_warp_ndim = Image(img_src.data[:, :], hdr=img_warp.hdr)
-                elif dim in ['3', '4']:
-                    img_warp_ndim = Image(img_src.data[:, :, :], hdr=img_warp.hdr)
```

Comment:
Note that this change here is a small refactor of the logic, since instead of trying to extract a relevant slice from `img_src` using `[:, :, :]` (which doesn't even work), we can just make a `ones` image with the dimensions of our choice, since all we're using `ImageCropper` for is masking.